src/atomate2/vasp/flows/phonons.py

```diff
@@ -0,0 +1,326 @@
+    born_maker: .BaseVaspMaker or None
+        Maker to compute the BORN charges.
+    phonon_displacement_maker : .BaseVaspMaker or None
+        Maker used to compute the forces for a supercell.
+    generate_frequencies_eigenvectors_kwargs : dict
+        Keyword arguments passed to :obj:`generate_frequencies_eigenvectors`.
+    create_thermal_displacements: bool
+        Bool that determines if thermal_displacement_matrices are computed
+    kpath_scheme: str
+        scheme to generate kpoints. Please be aware that
+         you can only use seekpath with any kind of cell
+         Otherwise, please use the standard primitive structure
+         Available schemes are:
+         "seekpath", "hinuma", "setyawan_curtarolo", "latimer_munro",
+         "all". "seekpath" and "hinuma" are the same definition but
+         seekpath can be used with any kind of unit cell as
+         it relies on phonopy to handle the relationship
+         to the primitive cell and not pymatgen
+    code: str
+        determines the dft code. currently only vasp is implemented.
+        This keyword might enable the implementation of other codes
+        in the future
+    """
+
+    name: str = "phonon"
+    sym_reduce: bool = True
+    symprec: float = 1e-4
+    displacement: float = 0.01
+    min_length: float | None = 20.0
+    prefer_90_degrees: bool = True
+    get_supercell_size_kwargs: dict = field(default_factory=dict)
+    use_primitive_standard_structure: bool = False
+    use_conventional_standard_structure: bool = False
+    bulk_relax_maker: BaseVaspMaker | None = field(
+        default_factory=lambda: DoubleRelaxMaker.from_relax_maker(TightRelaxMaker())
+    )
+    static_energy_maker: BaseVaspMaker | None = field(default_factory=StaticMaker)
+    generate_phonon_displacements_kwargs: dict = field(default_factory=dict)
+    run_phonon_displacements_kwargs: dict = field(default_factory=dict)
+    born_maker: BaseVaspMaker | None = field(default_factory=DielectricMaker)
+    phonon_displacement_maker: BaseVaspMaker = field(
+        default_factory=PhononDisplacementMaker
+    )
+    create_thermal_displacements: bool = True
+    generate_frequencies_eigenvectors_kwargs: dict = field(default_factory=dict)
+    kpath_scheme: str = "seekpath"
+    code: str = "vasp"
+
+    def make(
+        self,
+        structure: Structure,
+        prev_vasp_dir: str | Path | None = None,
+        born_manual: List[Matrix3D] | None = None,
+        full_born: bool = True,
+        epsilon_static_manual: Matrix3D | None = None,
+        total_dft_energy: float | None = None,
+        supercell_matrix: Matrix3D | None = None,
+    ):
+        """
+        Make flow to calculate the phonon properties.
+
+        Parameters
+        ----------
+        structure : .Structure
+            A pymatgen structure.
+        prev_vasp_dir : str or Path or None
+            A previous vasp calculation directory to use for copying outputs.
+        born_manual: Matrix3D
```

Comment:
Is it possible to remove the `full_born` parameter and just figure out with the full or reduced born has been passed by checking the length of the born array. E.g., using some pseudocode like:

```python
if len(born) == len(structure):
    full_born = False
```

This will have to be put inside the job where the born is actually used as the structure might not be available otherwise.

Also, if you go down this route, you can specify that either the reduced or full born description can be supplied, in the born parameter docstring.

Reply:
Thank for the suggestion.

Reply:
Will add the following and test if it works: 
`if len(born) == len(structure):
    full_born = True`